src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/LongPollingTransport.cs

```diff
@@ -30,10 +30,18 @@ public LongPollingTransport(CancellationToken timeoutToken, PipeReader applicati
         {
             try
             {
+                var headersFlushed = false;
+                if (context.Items.TryGetValue("FirstRequest", out var firstRequest))
+                {
+                    if ((bool)firstRequest) {
+                        await context.Response.Body.FlushAsync();
+                        headersFlushed = true;
+                    }
+                }
                 var result = await _application.ReadAsync(token);
                 var buffer = result.Buffer;
 
-                if (buffer.IsEmpty && result.IsCompleted)
+                if (buffer.IsEmpty && result.IsCompleted && !headersFlushed)
```

Comment:
Put a comment here saying that if we get an empty&completed result in the first request, we'll have to send a 200 because we've already flushed the headers, but that's OK since the client will poll again and immediately get the 204.